lib/xcov/manager.rb

```diff
@@ -219,11 +219,36 @@ def process_xcresults!(xcresult_paths)
             UI.important("Copying .xccovreport to #{path}") 
           end
 
+          if archive_paths.length > 1 && report_paths.length == archive_paths.length
+            Dir.chdir(output_path) do # xccov fails if ran from outside the output directory
```

Comment:
i recommend not to have restrictions around path from where xcov can be run.. since `ci` executions happens mainly from project `fastlane` directory.. instead how about below code and i have tested this below changes on my local

> if archive_paths.length > 1 && report_paths.length == archive_paths.length
            xcresult_file = File.absolute_path(xcresult_path)
            xcov_output_path = File.absolute_path(output_path)
            xcresult_formatted_mtime = File.mtime(xcresult_file).getlocal.strftime('%Y.%m.%d_%H-%M-%S-%z') # trying to mimic xcresult name format, but doesn't need to be
            merged_report_filename = "merged-#{xcresult_formatted_mtime}.xccovreport" # more unique name protects against multiple xcresult files being processed
            File.delete(merged_report_filename) if File.exist?(merged_report_filename) # xccov fails if the file already exists

            xccov_merged_archive_name = 'merged.xccovarchive' # directory created by `xccov merge` -> this name is known empirically
            FileUtils.rm_r(xccov_merged_archive_name) if File.exist?(xccov_merged_archive_name) # xccov fails if the directory already exists
            merged_archive_name = "merged-#{xcresult_formatted_mtime}.xccovarchive" # more unique name protects against multiple xcresult files being processed
            FileUtils.rm_r(merged_archive_name) if File.exist?(merged_archive_name) # xccov fails if the directory already exists

            archive_filenames = archive_paths.map { |path| "#{xcov_output_path}/#{File.basename(path)}".shellescape }
            report_filenames = report_paths.map { |path| "#{xcov_output_path}/#{File.basename(path)}".shellescape }
            cmd = "xcrun xccov merge --outReport #{xcov_output_path}/#{merged_report_filename} #{report_filenames.zip(archive_filenames).join(' ')}"
            FastlaneCore::CommandExecutor.execute(command: cmd)

            if File.exist?(xccov_merged_archive_name)
              File.rename(xccov_merged_archive_name, merged_archive_name)
            end
            report_paths = [File.join(output_path, merged_report_filename)]
          end

Reply:
hi @pradyhk, thanks a lot for the feedback! so looking at your code, it seems you avoid the temporary `chdir` by using absolute paths everywhere? doesn't sound bad at all to me, just more verbose, so I'd like to understand better the issue with using `chdir` in this case where it's just a temporary change

> The original working directory is restored when the block exits
https://ruby-doc.org/core-2.6.3/Dir.html#method-c-chdir

Reply:
use case where it was not working for me if
xcresult directory is -> `ios/enterprise/derivedData/Logs/Test`
xcov output directory is -> `ios/enterprise/Build/xcov`
fastlane directory from where xcov is invoked in Fastfile ->`ios/enterprise/fastlane/Fastfile`

Error : No such file or directory since `xcresult_file` will have the value `ios/enterprise/Build/xcov/derivedData/Logs/Test` which

Reply:
for the `xcresult_file` I'm using its absolute path so I'm not seeing how you got it to be evaluated as being inside `ios/enterprise/Build/xcov/derivedData/Logs/Test`

the environment I'm using for testing this also has the xcresult files in `DerivedData/Logs/Test` and the xcov output_path somewhere else, but all files that xcov generates (archive_paths, report_paths and the new merged stuff) go inside that output_path directory which is where we've `chdir` into

do you mind pasting your xcov configuration? the lines in your Fastfile or Dangerfile where you call it. Thanks again, looking forward to fixing that! 🙏

Reply:
```
+-----------------------------+----------------------------------------------------------------------------------------------------+
|                                                      Summary for xcov 1.7.0                                                      |
+-----------------------------+----------------------------------------------------------------------------------------------------+
| scheme                      | Enterprise                                                                                         |
| workspace                   | Enterprise.xcworkspace                                                                             |
| skip_slack                  | true                                                                                               |
| derived_data_path           | ./derivedData                                                                                      |
| include_targets             | Enterprise.app                                                                                     |
| minimum_coverage_percentage | 20.0                                                                                               |
| markdown_report             | true                                                                                               |
| json_report                 | true                                                                                               |
| output_directory            | ./Build/xcov/                                                                                      |
| html_report                 | true                                                                                               |
| slack_username              | xcov                                                                                               |
| slack_message               | Your *xcov* coverage report                                                                        |
| ignore_file_path            | ./fastlane/.xcovignore                                                                             |
| include_test_targets        | false                                                                                              |
| only_project_targets        | false                                                                                              |
| disable_coveralls           | false                                                                                              |
| ideFoundationPath           | /Applications/Xcode.app/Contents/Developer/../Frameworks/IDEFoundation.framework/Versions/A/IDEFo  |
|                             | undation                                                                                           |
| legacy_support              | false                                                                                              |
+-----------------------------+----------------------------------------------------------------------------------------------------+

[18:03:29]: Copying .xccovarchive to ./Build/xcov/action_0.xccovarchive
[18:03:29]: Copying .xccovarchive to ./Build/xcov/action_1.xccovarchive
[18:03:29]: Copying .xccovreport to ./Build/xcov/action_0.xccovreport
[18:03:29]: Copying .xccovreport to ./Build/xcov/action_1.xccovreport
No such file or directory @ rb_file_s_mtime - /Users/ios/enterprise/Build/xcov/derivedData/Logs/Test/hk2.xcresult
[18:03:29]: Error occured while exporting xccovreport from xcresult './derivedData/Logs/Test/hk2.xcresult'
[18:03:29]: Make sure you have both Xcode 11 selected and pointing to the correct xcresult file
+------------------+-------+
|       Lane Context       |
+------------------+-------+
| DEFAULT_PLATFORM | ios   |
| PLATFORM_NAME    |       |
| LANE_NAME        | lane1 |
+------------------+-------+
[18:03:29]: Failed to export xccovreport from xcresult

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | opt_out_usage              | 0           |
| 2    | Verifying fastlane version | 0           |
| 3    | default_platform           | 0           |
| 💥   | xcov                       | 0           |
+------+----------------------------+-------------+
```

Reply:
so apparently your issue comes from the fact that your configuration specifies relative paths instead of absolute paths for `derived_data_path ` and `output_directory`, I'd presume that to have unexpected results but also if you're seeing an issue just now, probably other people have already put effort into making that work everywhere else so I'll adjust the code here to work well with that. Thanks once again! I'll keep you posted

Reply:
@pradyhk just pushed a small change that works for me when simulating your scenario, do you mind telling us if it's okay now? 🙏

Reply:
changes lgtm... 👍 ..  i wont be able to test at the moment..

Reply:
it works for me on my local with latest comment .. thanks 💯

Reply:
awesome @pradyhk, great job on proactively checking this and being responsive, it's infinitely better to fix this now than later!